### PR TITLE
Update/add fade to long text in search component

### PR DESCRIPTION
**Pull request by budzanowski**
Wwhen input text is longer then the available width of input field the visible part of text is cut and can create the expression that it is all there is. 

no_fade

Now there is a fade that suggests that the text is longer then the visible part.

fade

To test open `/design` and write in search fields more charters then the width of search field.

Test live: https://calypso.live/?branch=update/add-fade-to-long-text-in-search-component

### CODE REVIEW

**Comment by folletto** [comment]
Tested on: Chrome, Firefox, Safari, Edge, IE11. 👍  

Works.

Since it's a component-level change, I also looked at other instances of Search and all seems good:

screen shot 2016-08-09 at 11 40 09

File: client/components/search/index.jsx [hunk 1]
```diff
@@ -271,15 +272,23 @@ const Search = React.createClass( {
...
 			rtl: this.props.dir === 'rtl',
 			ltr: this.props.dir === 'ltr',
 			search: true
 		} );
 
+		const inputClass = classNames ( {
+			'search__input': true,
+			'no-close-button': this.props.hideClose,
+			rtl: this.props.dir === 'rtl',
+			ltr: this.props.dir === 'ltr',
```
Comment (ockham) [inline client/components/search/index.jsx]: Why do we need to duplicate the `rtl` and `ltr` classes here? (Asking since they're already on L280-281.)

Comment (ockham) [inline client/components/search/index.jsx]: Also, can't we just

```js
const inputClass = classNames( this.props.dir, { /*...*/ } );
```

?

File: client/components/search/index.jsx [hunk 2]
```diff
@@ -313,22 +322,32 @@ const Search = React.createClass( {
...
-				aria-controls={ 'search-component-' + this.state.instanceId }
-				aria-label={ i18n.translate( 'Close Search', { context: 'button label' } ) }>
-			<Gridicon icon="cross" className={ 'search-close__icon' + ( this.props.dir ? ' ' + this.props.dir : '' ) } />
-			</span>
-		);
+		const gridIconClass = classNames ( {
+			'search-close__icon': true,
+			rtl: this.props.dir === 'rtl',
+			ltr: this.props.dir === 'ltr',
+		} );
```
Comment (ockham) [inline client/components/search/index.jsx]: https://github.com/Automattic/wp-calypso/pull/7351/files#r74422075

Comment (ockham) [inline client/components/search/index.jsx]: And given `'search-close__icon': true`, actually

```js
const gridIconClass = classNames ( 'search-close__icon', this.props.dir );
```

Maybe in combination with changing `dir`'s `propType` to 

```js
PropTypes.oneOf( ['ltr', 'rtl'] )
```

**Comment by ockham** [comment]
Left a few stylistic nits, but overall looks very good!

One question tho, you don't mention the addition of the `hideClose` prop in the PR desc, and it's apparently not used anywhere yet. Are we going to need that for our M5?

**Pull request closed without merging**